p2p/host/basic/basic_host.go

```diff
@@ -23,6 +23,8 @@ var log = logging.Logger("basichost")
 
 var NegotiateTimeout = time.Second * 60
 
+type AddrsFactory func([]ma.Multiaddr) []ma.Multiaddr
```

Comment:
Document this type please. Why it is here? What is it used for and maybe a practical example of how it can be used.